### PR TITLE
Fix for hanging Promise when calling quit() in disconnect

### LINKED ISSUE #42: Unresolved Promise when disconnecting from Redis due to expired connection
Hi @kjellmorten 

We have encountered an issue in our integration apps using `integreat-transporter-redis` that's related to the code that tries to disconnect from Redis when a connection has been open for more than the expiry timeout.
The issue manifested itself when DigitalOcean changed a bug in their hosted Redis service that had been keeping connections open for to long (potentially indefinitely). After they fixed the bug the recursive, setTimeout based "scheduler" in on of our apps froze after 10min, which is coincidentally 2x the 5min timeout that DigitalOcean had introduced.

We have been able to track it down to being cause by the Promise returned from the [connection.redisClient.quit()](https://github.com/integreat-io/integreat-transporter-redis/blob/main/src/disconnect.ts#L11) call in disconnect.ts never resolving if the connection to the Redis server is already closed due to the timeout on the server side. This does sound a bit like a bug in the `node-redis` library, but we haven't found any reports on it in their repo though.

We have however made a PR #43  with an attempt of mitigating the problem by adding a timeout around the redisClient.quit() function call. Please take a look at the PR and tell us what you think.

This bug is pretty critical for us as we're forced to restart the apps in question every 10 minutes (!) to reset the Redis connection

### PR DESCRIPTION
**Pull request by torbjornvatn**
It turns out that the Promise returned by redisClient.quit() neverresolves if the connection to Redis for some reson has been closed. To mitigate this we both try to check whether the connection isstill ready by checking redisConnection.isReady and also add aPromise.race based time out to the quit() function it self.

We have added a separate reconnect.test.ts that attempts to emulate that the connection to the Redis server closed when the disconnect code tries to call the quit() function.

Closes #42

### CODE REVIEW

**Comment by torbjornvatn** [comment]
Here's an issue in `node-redis` that about the same situation https://github.com/redis/node-redis/issues/2723

It mentions that [QUIT is deprecated](https://redis.io/docs/latest/commands/quit/), but that there is no good alternative yet. There's a safer [close()](https://github.com/redis/node-redis/blob/master/packages/client/lib/client/index.ts#L1074) coming in v5 of the client though.

[This](https://gist.github.com/JonCole/925630df72be1351b21440625ff2671f#file-redis-bestpractices-node-js-md) mentions that adding a `setInterval` triggered `ping()` to keep the connection open, but that seem a bit hacky to me.

File: src/tests/reconnect.test.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    const networkId = network?.Id ?? 'N/A'
+
+    await docker
+      .getNetwork(networkId)
+      .disconnect({ Container: 'reconnect-redis' })
+
+    // Wait for the connection to expire which should trigger an attempt to
+    // disconnect (and client.quit() or client.disconnect() under the hood)
+    // and then reconnect on the next call to transporter.connect()
+    clock.tick(options.connectionTimeout + 1)
```
Comment (kjellmorten) [inline src/tests/reconnect.test.ts:49]: I wonder if this is really doing anything here, as it is not waiting, just changing the time. I might misunderstand the intention, @torbjornvatn, but won't the `setTimeout()` just below be enough?

**Comment by kjellmorten** [comment]
Thanks for a good PR with very thorough explanations, @torbjornvatn!

I think this make sense, although I find it a bit hard to verify. I've published this as a beta release now, as this may make it easier to test. Is it possible for you to try and use version `1.3.0-beta.0` and see if that solves your issue?

Btw, the tests won't run, as the GitHub set up is not using `docker-compose.yml` for spinning up Redis in test. I've changed that, but can't commit to this PR, so I've cloned it and made changes as #44. Here I also changed how you call `PING`, to use a `SERVICE` action, to conform with other Integreat transporters, but the effect is the same.

**Comment by torbjornvatn** [comment]
> Thanks for a good PR with very thorough explanations, @torbjornvatn!
> 
> I think this make sense, although I find it a bit hard to verify. I've published this as a beta release now, as this may make it easier to test. Is it possible for you to try and use version `1.3.0-beta.0` and see if that solves your issue?
> 
> Btw, the tests won't run, as the GitHub set up is not using `docker-compose.yml` for spinning up Redis in test. I've changed that, but can't commit to this PR, so I've cloned it and made changes as #44. Here I also changed how you call `PING`, to use a `SERVICE` action, to conform with other Integreat transporters, but the effect is the same.

We can indeed test the beta version. Will report back on how it works.

**Comment by torbjornvatn** [comment]
We have tested this for some days in prod now, and it looks like it's working like a dream. 
The quit function timeout kicks in from time to time, but then it reconnects and our app keeps on processing

**Comment by torbjornvatn** [comment]
Superseeded by #44

**Pull request closed without merging**